### PR TITLE
Feature/aspect logging and local dev server

### PR DESCRIPTION
**Pull request by bbortt**
This PR contains the following changes:
* Allow `POST` aand `GET` requests from the client via `axios`.
* Use AOP Logging for the API
* Variable frontend configuration via `.env` or environment variables

### CODE REVIEW

File: src/main/kotlin/io/github/bbortt/buessle/app/configuration/AspectConfiguration.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import io.github.bbortt.buessle.app.aspect.ApiLoggingAspect
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+
+@Configuration
+class AspectConfiguration {
+
+    @Bean
+    fun apiLoggingAspect(): ApiLoggingAspect {
+        return ApiLoggingAspect();
```
Comment (linovifian) [inline src/main/kotlin/io/github/bbortt/buessle/app/configuration/AspectConfiguration.kt]: Do we need this? Couldn't you just annotate the `ApiLogginAspect` as a `@Component`?

File: src/main/kotlin/io/github/bbortt/buessle/app/configuration/DevCorsConfiguration.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        fun corsConfiguration(): UrlBasedCorsConfigurationSource {
+            val configuration = CorsConfiguration()
+            configuration.allowedOrigins = Collections.singletonList("http://localhost:3000")
+            configuration.allowedMethods = Arrays.asList("GET", "POST")
+            configuration.allowedHeaders = Collections.singletonList("content-type")
+            val source = UrlBasedCorsConfigurationSource()
+            source.registerCorsConfiguration("/api/**", configuration)
+            return source
+        }
+    }
```
Comment (linovifian) [inline src/main/kotlin/io/github/bbortt/buessle/app/configuration/DevCorsConfiguration.kt]: I suggest to not make the `DevCorsConfiguration` implement the `CorsFilter` but rather add another `@Bean` method. That way you could also remove the `companion object`

```suggestion
class DevCorsConfiguration {

    @Bean
    fun corsConfiguration(): UrlBasedCorsConfigurationSource {
        val configuration = CorsConfiguration()
        configuration.allowedOrigins = Collections.singletonList("http://localhost:3000")
        configuration.allowedMethods = Arrays.asList("GET", "POST")
        configuration.allowedHeaders = Collections.singletonList("content-type")
        val source = UrlBasedCorsConfigurationSource()
        source.registerCorsConfiguration("/api/**", configuration)
        return source
    }
    
    @Bean
    fun corsFilter(): CorsFilter = CorsFilter(corsConfiguration())
```

**Review by linovifian — CHANGES_REQUESTED** [review]

File: src/main/kotlin/io/github/bbortt/buessle/app/configuration/DevCorsConfiguration.kt (hunk 2 above)
Comment (bbortt) [reply src/main/kotlin/io/github/bbortt/buessle/app/configuration/DevCorsConfiguration.kt]: Valid point 👍

File: src/main/kotlin/io/github/bbortt/buessle/app/configuration/AspectConfiguration.kt (hunk 1 above)
Comment (bbortt) [reply src/main/kotlin/io/github/bbortt/buessle/app/configuration/AspectConfiguration.kt]: Hm yes, valid point. I am not a great fan of `@Component`s as they are "kind of hidden", but if we keep them all in `aspect` package we're good to go. I'll commit it later.

**Review by linovifian — APPROVED** [review]

**Pull request merged**